pkg/cli/state/vmclarity.go

```diff
@@ -407,3 +415,24 @@ func NewVMClarityState(client *backendclient.BackendClient, id AssetScanID) (*VM
 		assetScanID: id,
 	}, nil
 }
+
+func appendEffectiveScanConfigAnnotation(annotations *models.Annotations, config *families.Config) (*models.Annotations, error) {
+	configJSON, err := json.Marshal(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal effective families config: %w", err)
+	}
+	effectiveScanConfigAnnotations := models.Annotations{
+		{
+			Key:   utils.PointerTo(effectiveScanConfigAnnotationKey),
+			Value: utils.PointerTo(string(configJSON)),
+		},
+	}
+
+	if annotations == nil {
+		return &effectiveScanConfigAnnotations, nil
+	}
+	newAnnotations := *annotations
+	newAnnotations = append(newAnnotations, effectiveScanConfigAnnotations...)
+
+	return &newAnnotations, nil
+}
```

Comment:
We might need to check if the effectiveScanConfigAnnotiationKey is already in the list of annotations, and if it is overwrite it. It shouldn't happen in the normal flow of things but its worth guarding against.

Reply:
https://github.com/openclarity/vmclarity/pull/739/commits/fab1edc2a4ed52338bc246c3df3fb78d7b897165